transformation/src/main/scala/org/broadinstitute/monster/hca/HcaPipelineBuilder.scala

```diff
@@ -262,9 +265,21 @@ object HcaPipelineBuilder extends PipelineBuilder[Args] {
   def encode(msg: Msg): String =
     upack.transform(msg, StringRenderer()).toString
 
-  def validateJson(filenamesAndMsg: SCollection[(String, Msg)]): SCollection[(String, Msg)] = {
+  def logSchemaValidationError(filepath: String, errorMessage: String): Unit = {
+    val filenameRegex = "[^/]*$".r // match everything that is not followed by a "/" (only the filename)
+    val filename = filenameRegex.findFirstIn(filepath).getOrElse("")
+    val errorLog = ujson.Obj(
+      "errorType" -> ujson.Str("SchemaValidationError"),
+      "filePath" -> ujson.Str(filepath),
```

Comment:
This `filepath` value is coming from [here](https://github.com/DataBiosphere/hca-ingest/blob/master/transformation/src/main/scala/org/broadinstitute/monster/hca/HcaPipelineBuilder.scala#L109) - which the Apache Beam [documentation](https://beam.apache.org/releases/javadoc/2.1.0/org/apache/beam/sdk/io/fs/ResourceId.html) describes as _the name of the file or directory denoted by this ResourceId. The file name is the farthest element from the root in the directory hierarchy_.

I still can't tell whether it would return the full GCS path or just the part we want, but I don't know of a good way to to test it either than just running the workflow and looking at the logs. Let me know if you have any suggestions.

Reply:
I think we should test this before merging it, and unfortunately we don't have a good way set up to do that (I have an idea but it's a longer-term solution). For a quick fix, Sarah, I'll work with you today to coordinate how to test this. We'll make some dummy input with an invalid schema and run this in dataflow.